### PR TITLE
Add aria-label to page jump

### LINKED ISSUE #33075: [Issue] Add aria-label to page jump
This issue is automatically created based on existing pull request: magento/magento2#33067: Add aria-label to page jump

---------

### Description (*)
Issue : 2.4.4 Link Purpose (In Context))

Summary: The same link text, `...` is used for links going to different page jumps. Users might not know the difference between each `...` if there is no explanation

Fix: Add an 'aria-label' which indicates the page the user is skipping to

### Related Pull Requests
N/A
### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by bradleybrecher**
### Description (*)
Issue : 2.4.4 Link Purpose (In Context))

Summary: The same link text, `...` is used for links going to different page jumps. Users might not know the difference between each `...` if there is no explanation

Fix: Add an 'aria-label' which indicates the page the user is skipping to

### Related Pull Requests
N/A
### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#33075: Add aria-label to page jump

### CODE REVIEW

**Comment by bradleybrecher** [comment]
I will not make a test for this addition, so someone feel free to add it

**Comment by gabrieldagama** [comment]
@magento create issue

File: app/code/Magento/Theme/view/frontend/templates/html/pager.phtml [hunk 1]
```diff
@@ -87,7 +88,10 @@
 
             <?php if ($block->canShowNextJump()): ?>
                 <li class="item">
-                    <a class="page next jump" title="" href="<?= $block->escapeUrl($block->getNextJumpUrl()) ?>">
+                    <a class="page next jump"
+                       title=""
+                       aria-label="Skip to page <?= $block->escapeHtml($block->getNextJumpPage()) ?>"
```
Comment (blmage) [inline app/code/Magento/Theme/view/frontend/templates/html/pager.phtml]: Can you translate this?

Comment (bradleybrecher) [reply app/code/Magento/Theme/view/frontend/templates/html/pager.phtml]: @blmage can you clarify further what you mean by "translate this"?

Comment (Den4ik) [reply app/code/Magento/Theme/view/frontend/templates/html/pager.phtml]: @bradleybrecher 'Skip to Page' should be translatable with `__()`

Comment (bradleybrecher) [reply app/code/Magento/Theme/view/frontend/templates/html/pager.phtml]: change pushed & ready for another review

**Comment by Den4ik** [comment]
@magento run all tests

File: app/code/Magento/Theme/view/frontend/templates/html/pager.phtml [hunk 2]
```diff
@@ -61,6 +61,7 @@
                 <li class="item">
                     <a class="page previous jump"
                        title=""
+                       aria-label="<?= $block->escapeHtml(__('Skip to page %1', $block->getNextJumpPage())) ?>"
```
Comment (Den4ik) [inline app/code/Magento/Theme/view/frontend/templates/html/pager.phtml]: Please use `$escaper->escapeHtmlAttr()`
https://github.com/magento/magento2/blob/2.4-develop/lib/internal/Magento/Framework/Escaper.php#L259

File: app/code/Magento/Theme/view/frontend/templates/html/pager.phtml [hunk 3]
```diff
@@ -87,7 +88,10 @@
 
             <?php if ($block->canShowNextJump()): ?>
                 <li class="item">
-                    <a class="page next jump" title="" href="<?= $block->escapeUrl($block->getNextJumpUrl()) ?>">
+                    <a class="page next jump"
+                       title=""
+                       aria-label="<?= $block->escapeHtml(__('Skip to page %1', $block->getNextJumpPage())) ?>"
```
Comment (Den4ik) [inline app/code/Magento/Theme/view/frontend/templates/html/pager.phtml]: Please use `$escaper->escapeHtmlAttr()`
https://github.com/magento/magento2/blob/2.4-develop/lib/internal/Magento/Framework/Escaper.php#L259

**Review by Den4ik — CHANGES_REQUESTED** [review]
@bradleybrecher Thanks for the update. Please look at my comments

File: app/code/Magento/Theme/view/frontend/templates/html/pager.phtml (hunk 3 above)
Comment (bradleybrecher) [reply app/code/Magento/Theme/view/frontend/templates/html/pager.phtml]: Updated, also added docblock

**Comment by Den4ik** [comment]
@magento run all tests

**Review by Den4ik — APPROVED** [review]
✅ Approved

In my opinion all looks ok and don't have any risks

**Comment by magento-engcom-team** [comment]
Hi @Den4ik, thank you for the review.
[ENGCOM-9108](https://jira.corp.magento.com/browse/ENGCOM-9108) has been created to process this Pull Request

**Comment by sdzhepa** [comment]
@magento run all tests

**Comment by gabrieldagama** [comment]
@magento run all tests

**Comment by engcom-Alfa** [comment]
Hello @bradleybrecher and @Den4ik
Thank you for your contribution and collaboration!
I am a new QA engineer and would like to test this Pull Request to unblock and speed up the delivery. 
Unfortunately, with the current description, I do not see how it can be tested.
Could you please provide more details on how it can be tested?
For example
- Steps to reproduce and actual/expected result. That I can use to check previous and current behavior to confirm it is fixed.
- Functional user flows that can be affected by such changes
- Any other advice/ recommendations that will allow us to test it
Thank you in advance!

**Comment by sdzhepa** [comment]
@magento run all tests

**Comment by engcom-Alfa** [comment]
@magento run Unit Tests

**Comment by engcom-Alfa** [comment]
Hi @bradleybrecher , @Den4ik & @blmage 
Thanks for the collaboration & contribution!

:x: QA not Passed

_Anchor links with text `...` are not containing any labels. So required to add aria-label attribute_

**Manual Scenario Steps**

1. Install Magento 2.4-develop instance with sample data.
2. Make sure to have any category with more than 12 products at least to see the next pagination link.
3. Admin side- Change the settings to have 3 dots for products next page navigation in the front end.
4. View the products list page in the front end and validate the next page `...` (3 dots) anchor link


**:heavy_check_mark: Expected result After Fix**
- Expected to have a _aria-label_ attribute in the DOM to the link.


**:x: Actual result After Fix** 
- There is no _aria-label_ attribute in the DOM to the link. Below is the screenshot of it.
- image

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@engcom-Alfa I don't know how you tests it, but **QA Passed for me**

**Actual Result**
2021-08-13-12-47-16@2x

**Review by Den4ik — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @Den4ik, thank you for the review.
[ENGCOM-9108](https://jira.corp.magento.com/browse/ENGCOM-9108) has been created to process this Pull Request

**Comment by Den4ik** [comment]
@engcom-Alfa Looks like I understood.
Looks like you set `...` to `Anchor Text for Previous` and `Anchor Text for Next`

Please follow little bit different scenario:
1. Set `Pagination Frame Skip` to 1

**Comment by engcom-Alfa** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Alfa** [comment]
@magento run Functional Tests CE

**Comment by engcom-Alfa** [comment]
@magento run Functional Tests EE

**Pull request merged**